Googletest build failed on Gradescope

Hi there,

I failed to build Googletest on Gradescope, which I believe is an error on the server side. I test a previous submission that was successfully built on Gradescope but it fails now. The error message is as below: 


/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest.h:1352:11: error: The left operand of '==' is a garbage value [clang-analyzer-core.UndefinedBinaryOperatorResult,-warnings-as-errors]
  if (lhs == rhs) {
          ^
/autograder/bustub/test/storage/grading_tmp_tuple_page_test.cpp:101:3: note: Control jumps to 'case 0:'  at line 101
  ASSERT_EQ(*reinterpret_cast<page_id_t *>(data), page_id);
  ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest.h:1869:31: note: expanded from macro 'ASSERT_EQ'
#define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
                              ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest.h:1853:3: note: expanded from macro 'GTEST_ASSERT_EQ'
  ASSERT_PRED_FORMAT2(::testing::internal::EqHelper::Compare, val1, val2)
  ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest_pred_impl.h:148:3: note: expanded from macro 'ASSERT_PRED_FORMAT2'
  GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_FATAL_FAILURE_)
  ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest_pred_impl.h:134:3: note: expanded from macro 'GTEST_PRED_FORMAT2_'
  GTEST_ASSERT_(pred_format(#v1, #v2, v1, v2), on_failure)
  ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest_pred_impl.h:78:3: note: expanded from macro 'GTEST_ASSERT_'
  GTEST_AMBIGUOUS_ELSE_BLOCKER_                                 \
  ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/internal/gtest-port.h:644:3: note: expanded from macro 'GTEST_AMBIGUOUS_ELSE_BLOCKER_'
  switch (0)                          \
  ^
/autograder/bustub/test/storage/grading_tmp_tuple_page_test.cpp:101:3: note: Calling 'EqHelper::Compare'
  ASSERT_EQ(*reinterpret_cast<page_id_t *>(data), page_id);
  ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest.h:1869:31: note: expanded from macro 'ASSERT_EQ'
#define ASSERT_EQ(val1, val2) GTEST_ASSERT_EQ(val1, val2)
                              ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest.h:1853:23: note: expanded from macro 'GTEST_ASSERT_EQ'
  ASSERT_PRED_FORMAT2(::testing::internal::EqHelper::Compare, val1, val2)
                      ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest_pred_impl.h:148:23: note: expanded from macro 'ASSERT_PRED_FORMAT2'
  GTEST_PRED_FORMAT2_(pred_format, v1, v2, GTEST_FATAL_FAILURE_)
                      ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest_pred_impl.h:134:17: note: expanded from macro 'GTEST_PRED_FORMAT2_'
  GTEST_ASSERT_(pred_format(#v1, #v2, v1, v2), on_failure)
                ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest_pred_impl.h:79:52: note: expanded from macro 'GTEST_ASSERT_'
  if (const ::testing::AssertionResult gtest_ar = (expression)) \
                                                   ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest.h:1371:56: note: Passing value via 3rd parameter 'lhs'
    return CmpHelperEQ(lhs_expression, rhs_expression, lhs, rhs);
                                                       ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest.h:1371:12: note: Calling 'CmpHelperEQ<int, int>'
    return CmpHelperEQ(lhs_expression, rhs_expression, lhs, rhs);
           ^
/autograder/bustub/build/googletest-src/googletest/include/gtest/gtest.h:1352:11: note: The left operand of '==' is a garbage value
  if (lhs == rhs) {
          ^
The files that failed were:
[u'/autograder/bustub/test/storage/tmp_tuple_page_test.cpp',
 u'/autograder/bustub/test/storage/grading_tmp_tuple_page_test.cpp']
Note that a failing .h file will fail all the .cpp files that include it.


CMakeFiles/check-clang-tidy.dir/build.make:57: recipe for target 'CMakeFiles/check-clang-tidy' failed
make[3]: *** [CMakeFiles/check-clang-tidy] Error 1
CMakeFiles/Makefile2:164: recipe for target 'CMakeFiles/check-clang-tidy.dir/all' failed
make[2]: *** [CMakeFiles/check-clang-tidy.dir/all] Error 2
CMakeFiles/Makefile2:171: recipe for target 'CMakeFiles/check-clang-tidy.dir/rule' failed
make[1]: *** [CMakeFiles/check-clang-tidy.dir/rule] Error 2
Makefile:214: recipe for target 'check-clang-tidy' failed
make: *** [check-clang-tidy] Error 2

Any help would be appreciated. Thanks!

--- a/src/include/storage/page/tmp_tuple_page.h
+++ b/src/include/storage/page/tmp_tuple_page.h
@@ -20,7 +20,10 @@ namespace bustub {
  */
 class TmpTuplePage : public Page {
  public:
-  void Init(page_id_t page_id, uint32_t page_size) {}
+  void Init(page_id_t page_id, uint32_t page_size) {
+    memcpy(GetData(), &page_id, sizeof(page_id_t));
+    memcpy(GetData() + sizeof(page_id_t), &page_size, sizeof(uint32_t));
+  }
 
   page_id_t GetTablePageId() { return INVALID_PAGE_ID; }
 

